src/libelektra/CMakeLists.txt

```diff
@@ -68,18 +69,20 @@ if (BUILD_SHARED)
 	include_directories (${elektra-shared_INCLUDES})
 
 	set_target_properties (elektra PROPERTIES
-			COMPILE_DEFINITIONS HAVE_KDBCONFIG_H)
-	set_target_properties (elektra PROPERTIES
-			VERSION ${KDB_VERSION} SOVERSION ${SO_VERSION})
-	set_target_properties (elektra PROPERTIES CLEAN_DIRECT_OUTPUT 1)
+		COMPILE_DEFINITIONS "HAVE_KDBCONFIG_H"
+		VERSION ${KDB_VERSION} SOVERSION ${SO_VERSION}
+		CLEAN_DIRECT_OUTPUT 1
+	)
+	if (${LD_ACCEPTS_VERSION_SCRIPT})
+		set_target_properties (elektra PROPERTIES
+			LINK_FLAGS "-Wl,--version-script=${__symbols_file}"
+		)
+	endif ()
 
 	#search only in the prefix the user decided to install elektra
 	#(e.g. /usr/local/)
 	set (ELEKTRA_RPATH "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}/${TARGET_PLUGIN_FOLDER}")
-
-	#RPATH settings, see http://www.itk.org/Wiki/CMake_RPATH_handling
-	#those settings are needed to find the plugins at runtime even without ld.so.conf
```

Comment:
Please do not remove essential comments. RPATH is quite misunderstood and some thought its not needed.